conans/util/runners.py

```diff
@@ -32,6 +33,59 @@ def pyinstaller_bundle_env_cleaned():
         yield
 
 
+# Windows (and possibly other platforms) should have a clean environment,
+# unpolluted by anything in the user's environment.
+# Anything except the bare minimum should be added to the environment through
+# conan's environment mechanisms.
+# This whitelist could potentially be slimmed down further.
+def _env_for_Popen():
+    newenv = None
+    if platform.system() == 'Windows':
```

Comment:
The problem is the variables I'm having problems with are not things I'm adding.  It is bash that is creating !ExitCode multiple times in different cases.
This would affect all users.

I thought it should be conan's responsibility though... I thought it was conan's job to control the build environment as best it can...

I don't think recipes should be depending on environmental variables from outside the conan-defined environment.  The recipe, lock file, and profiles, should say everything.
If users want something added to their environment, add it to their profile.  Isn't that what it is for?

Reply:
I'll continue the discussion on the merits in the issue